Frameworks/UIKit/UIPasteboard.mm

```diff
@@ -464,50 +566,60 @@ + (NSString*)_getStringFromClipboard {
     return [stringData autorelease];
 }
 
-+ (void)_setStringToClipboard:(NSString*)stringData {
++ (NSString*)_getStringFromClipboard {
+    WADDataPackageView* wasDataPackageView = _getClipboardContent();
+    return [UIPasteboard _getStringFromDataPackageView:wasDataPackageView];
+}
+
++ (void)_setStringOnDataPackage:(NSString*)stringData DataPackage:(WADDataPackage*)dataPackage {
     NSData* data = [stringData dataUsingEncoding:NSUnicodeStringEncoding];
-    WADDataPackage* dataPackage = [[WADDataPackage make] autorelease];
+    [UIPasteboard _setData:(NSString*)kUTTypeText withData:data on:dataPackage];
+}
 
++ (void)_setStringToClipboard:(NSString*)stringData {
+    WADDataPackage* dataPackage = [[WADDataPackage make] autorelease];
     dataPackage.requestedOperation = WADDataPackageOperationCopy;
-    [UIPasteboard _setData:(NSString*)kUTTypeText withData:data on:dataPackage];
+    [UIPasteboard _setStringOnDataPackage:stringData DataPackage:dataPackage];
     [WADClipboard setContent:dataPackage];
+    [WADClipboard flush];
 }
 
 /**
  @Status Caveat
  @Notes Only general pasteboard, for other type, it just will return nil.
 */
 - (UIImage*)image {
-    if ([_name isEqualToString:UIPasteboardNameGeneral]) {
-        return [UIPasteboard _getImageFromClipboard];
+    if (![_name isEqualToString:UIPasteboardNameGeneral]) {
+        TraceWarning(TAG, L"Only general pasteboard supported");
+        return nil;
     }
 
-    return nil;
+    return [UIPasteboard _getImageFromClipboard];
 }
 
 /**
  @Status Caveat
  @Notes Only general pasteboard.
 */
 - (void)setImage:(UIImage*)aImage {
-    if (aImage == nil) {
+    if (![_name isEqualToString:UIPasteboardNameGeneral]) {
+        TraceWarning(TAG, L"Only general pasteboard supported");
         return;
     }
 
-    if ([_name isEqualToString:UIPasteboardNameGeneral]) {
+    if (aImage != nil) {
         [UIPasteboard _setImageToClipboard:aImage];
     }
 }
 
-+ (UIImage*)_getImageFromClipboard {
++ (UIImage*)_getImageFromDataPackageView:(WADDataPackageView*)dataPackageView {
```

Comment:
nevermind - i see that you use the dataPackageView before calling this method

---
In reply to: [87640217](https://github.com/Microsoft/WinObjC/pull/1358#discussion_r87640217) [](ancestors = 87640217)